src/Text.php

```diff
@@ -20,6 +20,8 @@ public static function beautify($markdown)
     {
         // remove whitespace on lines with no non-whitespace characters
         $markdown = preg_replace('/\n\s+\n/', "\n\n", $markdown);
+        $markdown = preg_replace('/^[^\S\n]+\n/', "\n", $markdown);
```

Comment:
Given how many new lines we deal with, would it be worth looking at the `\R` character? I found an interesting article about it here: https://nikic.github.io/2011/12/10/PCRE-and-newlines.html. It may be over complicating things though as I am sure `\n` will work in 99% of cases. Feel free to ignore.

Reply:
@nigelhanlon How does this look?

```diff
diff --git a/src/Text.php b/src/Text.php
index 1e9ad06..4e677c6 100644
--- a/src/Text.php
+++ b/src/Text.php
@@ -9,6 +9,8 @@ class Text implements CopilotTag
     {
         if (!is_string($text)) throw new \InvalidArgumentException("Text::__construct first argument \$text must be a string. Given: ".($text ? "$text " : "")."(".gettype($text).").");
         $this->text = $text;
+        // convert newline types to LF newlines
+        $this->text = preg_replace('/\R/', "\n", $this->text);
     }
 
     public function write()
```

Reply:
lgtm 👍

Reply:
(see [Text.php line 13](https://github.com/conde-nast-international/copilot-markdown-generator-php/pull/7/files#diff-25cc76266925609cf768caf23952e0deR13))